bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/obscomp/SPSiteQuality.java

```diff
@@ -103,6 +104,10 @@ public long getStart() {
             return start;
         }
 
+        public Option<Instant> getEnd() {
+            return this.repeat == -1 ? None.instance() : new Some<>(Instant.ofEpochMilli(this.start + this.repeat * this.period + this.duration));
+        }
```

Comment:
A couple notes on this:

* Compare the repeat count to `TimingWindow.REPEAT_FOREVER` instead of directly `-1` so that it is clearer what is going on
* If the `duration == TimingWindow.WINDOW_REMAINS_OPEN_FOREVER` (which is just `-1`), then there will also be no end